core/templates/dev/head/domain/collection/search-explorations-backend-api.service.ts

```diff
@@ -16,38 +16,52 @@
  * @fileoverview Service to search explorations metadata.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-require('services/alerts.service.ts');
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
 
-angular.module('oppia').factory('SearchExplorationsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'SEARCH_EXPLORATION_URL_TEMPLATE',
-  function(
-      $http, $q, UrlInterpolationService,
-      SEARCH_EXPLORATION_URL_TEMPLATE) {
-    var _fetchExplorations = function(
-        searchQuery, successCallback, errorCallback) {
-      var queryUrl = UrlInterpolationService.interpolateUrl(
-        SEARCH_EXPLORATION_URL_TEMPLATE, {
-          query: btoa(searchQuery)
-        }
-      );
-      $http.get(queryUrl).then(function(response) {
-        successCallback(response.data);
-      }, function(errorResponse) {
-        errorCallback(errorResponse.data);
-      });
-    };
-    return {
-      /**
-       * Returns exploration's metadata dict, given a search query. Search
-       * queries are tokens that will be matched against exploration's title
-       * and objective.
-       */
-      fetchExplorations: function(searchQuery) {
-        return $q(function(resolve, reject) {
-          _fetchExplorations(searchQuery, resolve, reject);
-        });
+import { CollectionEditorPageConstants } from
+  'pages/collection-editor-page/collection-editor-page.constants';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SearchExplorationsBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolationService: UrlInterpolationService
+  ) {}
+
+  private _fetchExplorations(
+      searchQuery: string, successCallback: (
+      value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var queryUrl = this.urlInterpolationService.interpolateUrl(
+      CollectionEditorPageConstants.SEARCH_EXPLORATION_URL_TEMPLATE, {
```

Comment:
Hmm, this doesn't belong in `CollectionEditorPageConstants`. Can you move it to LibraryPageConstants?

Reply:
I didn't get -- it's present in `collection-editor-page.constants.ts`.

Reply:
yeah, I mean can you move that constant to libraryPageConstants? Since it's not really a collection editor thing.